src/osyris/spatial/subdomain.py

```diff
@@ -50,4 +55,8 @@ def extract_box(dataset, dx, dy, dz, origin):
         if np.any(c):
             subdomain[name] = dataset[name][c]
 
+    change_origin(subdomain, origin)
```

Comment:
Strange that extracting subdomain also re-centers now?

Reply:
I can't really think of a reason not to translate the coordinates after a subdomain extraction. Maybe if we introduce a default parameter to `origin`, but then I don't see the point of having your coordinates centered on the 0,0,0 of the Ramses grid.

Reply:
> so that they know where in the parent (global) box they are located.

That is why I added the `.origin` attribute to the dataset when changing origins, but the rest of your comment has indeed convinced me that I am restricting what users can do with the spatial submodule by forcing the translation.